File: chapter1/map.pl
#!/usr/bin/env perl
# Map function for hadoop streaming using Perl.

while (<STDIN>) {
    foreach my $word (split /\s+/, $_) {
        $word =~ tr/A-Z/a-z/;
        $word =~ s/[[:punct:]]//g;
        printf "%s\n", $word;
    }
}

File: chapter1/map.py
#!/usr/bin/env python
# Mapper function for Hadoop streaming using python
import sys, string
for line in sys.stdin:
    words = line.strip().split()
    for word in words:
        word = word.lower()
        word = word.translate(None, string.punctuation)
        print("%s\t%d" % (word, 1))
